File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n==0) return 0;
        if (n==1) return 1;
        if (n==2) return 1;
        
        
        vector<int> t(n+2,0);
        t[0] = 0;
        t[1] = 1;
        t[2] = 1;
        for (int i = 3;i<=n;i++){
            t[i] = t[i-1] + t[i-2] + t[i-3];
        }
        
        return t[n];
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        
        trib = [0 for _ in range(38)]
        trib[0] = 0
        trib[1] = 1
        trib[2] = 1
        
        
        if n==0 or n==1 or n==2:
            return trib[n]
        
        for i in range(3,n+1):
            trib[i] = trib[i-1] + trib[i-2] + trib[i-3]
            
            
        return trib[n]